core/commands/add.go

```diff
@@ -303,6 +352,23 @@ func addDir(n *core.IpfsNode, dir files.File, out chan interface{}, progress boo
 	return tree, nil
 }
 
+func checkForLocalIgnorePatterns(dir string, ignoreFilePatterns []ignore.GitIgnore) []ignore.GitIgnore {
+	var ignorePathname string
+	if dir == "." {
+		ignorePathname = ".ipfsignore"
+	} else {
+		ignorePathname = path.Join(dir, ".ipfsignore")
```

Comment:
I think you are right, that line was written before I had the folder level recursion working correctly. The check inside of `Run` should now cover this case.